### ISSUE TITLE
tox.ini, setup.cfg and .flake config files are wrongly ignored by flake8 CLI

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#734]
In GitLab by @mastacheata on May 13, 2020, 09:54

## *Please describe how you installed Flake8*

```
$ pip install flake8 flake8_formatter_junit_xml pep8-naming
```
(This is a GitLab throwaway runner, I'm aware this installs globally)

## *Please provide the exact, unmodified output of `flake8 --bug-report`*

```
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.7.7",
    "system": "Linux"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "naming",
      "version": "0.10.0"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.6.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.2.0"
    }
  ],
  "version": "3.8.1"
}
```

## *Please describe the problem or feature*  
Flake8 as of 3.8 ignores the tox.ini configuration. In 3.7 it seems to behave just fine.

## *If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*  
Given the following project structure:  
```
/project_folder
|backend
|- tox.ini
|- package
|-- __init__.py
|-- something.py
|frontend
|-...
```
(project_folder is /project_folder, backend is /project_folder/backend, etc.)  

and the following tox.ini contents (the exclude list is longer in reality, but this is just for sample purposes):  
```
[flake8]
max-line-length = 120
ignore-names=setUpTestData,setUp,setUpClass
exclude =
    .git,
    */migrations/*
```
and given that something.py has a line with 99 characters in it.  

Now when I run flake8 from the root (/) directory with the following commandline:  
`flake8 --format junit-xml backend/ | tee flake8.xml`  

### *What I expect to happen:*  
Flake8 ends without reporting any problems.

### *What happens instead:*  
Flake8 ends and reports `E501 line too long (102 > 79 characters)`  
The 79 characters seem to be the default value set by pycodestyle when no CLI- or config-file-option is specified.

The error persists if I move tox.ini one level down into the /project_folder/backend/ folder or when I rename it to setup.cfg or .flake8.  
The error only appears in Flake8 3.8.0 and 3.8.1  
It's not limited to the max-line-length option, Flake8 also ignores the exclusions given in the config file.  

## *Summary*:

To sum it up: Flake8 3.8 ignores tox.ini, setup.cfg and .flake8 inside of project folder(s).

**Comment by asottile** [comment]
In GitLab by @ericvw on May 13, 2020, 09:59

This is a duplicate of #639.

This behavior is an expected change (see https://gitlab.com/pycqa/flake8/-/merge_requests/363).

The [3.8.0 release notes](https://flake8.pycqa.org/en/latest/release-notes/index.html) mention this.  Going forward you either need to do a `cd <dir> && flake8 …` or `flake8 --config <path/to/config>`.

Apologies for being affected by this intentional behavior change.

**Comment by asottile** [comment]
In GitLab by @asottile on May 13, 2020, 10:00

closed
